File: chain-of-responsibility/typescript/jpg-handler.ts
import {BaseHandler} from "./base-handler";
import {Image} from "./image";

// JPG dosyalarını dönüştüren sınıf.
// UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
export class JPGHandler extends BaseHandler {
  public handleRequest(image: Image): void {
    if(image.extension === 'JPG') {
      console.log('JPG to PNG');
      // JPG işlemine ait dönüştürme kodları işlemleri.
    } else {
      // Bu sınıfa ait bir işlem değilse zincirin bir sonraki halkasına aktarılır.
      this.nextHandler.handleRequest(image);
    }
  }
}

File: chain-of-responsibility/go/jpg-handler.go
package main

import "fmt"

// JPG dosyalarını dönüştüren sınıf.
// UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
type jpgHandler struct {
	nextHandler BaseHandler
}

func NewJPGHandler() *jpgHandler {
	return &jpgHandler{}
}

func (h *jpgHandler) SetNextHandler(handler BaseHandler) {
	h.nextHandler = handler
}

func (h jpgHandler) HandleRequest(img Image) {
	if img.Extension == "JPG" {
		fmt.Println("JPG to PNG")
		// JPG işlemine ait dönüştürme kodları işlemleri.
	} else {
		// Bu sınıfa ait bir işlem değilse zincirin bir sonraki halkasına aktarılır.
		h.nextHandler.HandleRequest(img)
	}
}
